src/Microsoft.AspNet.Server.Kestrel/Http/SocketInput.cs

```diff
@@ -217,7 +217,10 @@ public void OnCompleted(Action continuation)
             else
             {
                 // THIS IS AN ERROR STATE - ONLY ONE WAITER CAN WAIT
-                throw new InvalidOperationException("Concurrent reads are not supported.");
```

Comment:
I wonder if we should also set `_awaitableError` so the exception gets surfaced to the reader.